### PR TITLE
add `self.use_ada_layer_norm_*` params back to `BasicTransformerBlock`

### PR DESCRIPTION
**Pull request by yiyixuxu**
fix https://github.com/huggingface/diffusers/issues/6838

I also fixed some other error in the sd reference community pipelines while I'm at it (so also fix https://github.com/huggingface/diffusers/issues/5028#issuecomment-1925483097)

we should deprecate these params too

### CODE REVIEW

File: examples/community/stable_diffusion_reference.py [hunk 1]
```diff
@@ -538,7 +538,7 @@ def hack_CrossAttnDownBlock2D_forward(
 
             return hidden_states, output_states
 
-        def hacked_DownBlock2D_forward(self, hidden_states, temb=None):
+        def hacked_DownBlock2D_forward(self, hidden_states, temb=None, **kwargs):
```
Comment (sayakpaul) [inline examples/community/stable_diffusion_reference.py:541]: Why is `kwargs` needed?

File: examples/community/stable_diffusion_reference.py [hunk 2]
```diff
@@ -634,7 +634,9 @@ def hacked_CrossAttnUpBlock2D_forward(
 
             return hidden_states
 
-        def hacked_UpBlock2D_forward(self, hidden_states, res_hidden_states_tuple, temb=None, upsample_size=None):
+        def hacked_UpBlock2D_forward(
+            self, hidden_states, res_hidden_states_tuple, temb=None, upsample_size=None, **kwargs
```
Comment (sayakpaul) [inline examples/community/stable_diffusion_reference.py:638]: Same.

File: src/diffusers/models/attention.py [hunk 3]
```diff
@@ -158,6 +158,12 @@ def __init__(
         super().__init__()
         self.only_cross_attention = only_cross_attention
 
+        self.use_ada_layer_norm_zero = (num_embeds_ada_norm is not None) and norm_type == "ada_norm_zero"
+        self.use_ada_layer_norm = (num_embeds_ada_norm is not None) and norm_type == "ada_norm"
+        self.use_ada_layer_norm_single = norm_type == "ada_norm_single"
+        self.use_layer_norm = norm_type == "layer_norm"
+        self.use_ada_layer_norm_continuous = norm_type == "ada_norm_continuous"
```
Comment (sayakpaul) [inline src/diffusers/models/attention.py:165]: I am okay with this.

File: examples/community/stable_diffusion_xl_reference.py [hunk 4]
```diff
@@ -603,7 +603,9 @@ def hacked_CrossAttnUpBlock2D_forward(
 
             return hidden_states
 
-        def hacked_UpBlock2D_forward(self, hidden_states, res_hidden_states_tuple, temb=None, upsample_size=None):
+        def hacked_UpBlock2D_forward(
+            self, hidden_states, res_hidden_states_tuple, temb=None, upsample_size=None, **kwargs
```
Comment (sayakpaul) [inline examples/community/stable_diffusion_xl_reference.py:607]: Same.

**Review by sayakpaul — APPROVED** [review]
Thanks for the swift action. Just a question.

File: examples/community/stable_diffusion_reference.py (hunk 1 above)
Comment (yiyixuxu) [reply examples/community/stable_diffusion_reference.py:541]: `hacked_DownBlock2D_forward` is a function they wrote to replace the `forward` method of the `DownBlock2D` ,hence the signature has to match. We added a new argument `scale` for the lora refactor and causes an error here  without `**kwargs`

I think they should write their custom blocks instead

File: src/diffusers/models/attention.py (hunk 3 above)
Comment (pcuenca) [reply src/diffusers/models/attention.py:165]: Sorry for the late reply. I'd maybe just add a comment to state that they are kept for back-compatibility reasons.

**Pull request merged**